Rakefile

```diff
@@ -10,8 +10,51 @@ namespace :design_system do
   end
 end
 
-desc "Rubocop Linting"
-task :rubocop do
-  puts "Running rubocop linter"
-  system("bundle exec rubocop") || raise
+desc "Run all checks in the local context"
+task check: ["ruby:lint", "npm:test"]
+
+namespace :ruby do
+  desc "Lint ruby files"
+  task lint: [:rubocop, :haml_lint]
+
+  desc "Rubocop Linting"
+  task :rubocop do
+    puts "Running rubocop linter"
+    system("bundle exec rubocop")
+  end
+
+  desc "Haml Linting"
+  task :haml_lint do
+    puts "Running haml-lint"
+    system("bundle exec haml-lint haml styleguide")
+  end
+end
+
+namespace :npm do
+  desc "Run node test"
+  task test: [:lint, :jest]
+
+  desc "Run jest tests in node"
+  task jest: :build do
+    puts "Running jest tests"
+    system("npm run jest")
+  end
+
+  desc "Run linting jobs in node"
+  task lint: :build do
+    puts "Running Node lint"
+    system("npm run lint")
+  end
+
+  desc "Install Node modules"
+  task :install do
+    puts "Installing modules"
+    system("npm install") || raise
+  end
+
+  desc "Build Node files"
```

Comment:
Minor OCD, order of these. Lets have `install` at the bottom as the one relied on by all others.

Reply:
Sure, easy to do.